src/components/Buttons/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,98 @@
+import { SetStateAction, useState } from "react";
+import globalStyle from "@/GlobalClasses.module.scss";
+import style from "./Carousel.module.scss";
+import ArrowBack from "@/components/Buttons/ArrowButtons/ArrowBack";
+import ArrowNext from "../ArrowButtons/ArrowNext";
+import ArrowButton from "@/components/Buttons/ArrowButtons/ArrowButton";
+import wheatImg from "@/assets/wheat.svg";
+
+interface CarouselProps {
+  data: string[];
+  registerIndex?: number;
+  isSwitched: boolean;
+  registerName: string;
+  initialValue: string | string[];
+  setValue: (name: string, value: SetStateAction<any>) => void;
+  register: (...args: any[]) => any;
```

Comment:
**nit:** There should not be any any - if you dont know what type should be here you can always use Typescript generics. However, I would recommend to not pass this register function as it just returns some typical HTMLInputElement props that you can declare explicitly